src/log_parser.py

```diff
@@ -13,29 +10,33 @@
 import pandas as pd
 
 class Parser(object):
-    header_fields = {}
+    header_fields = dict()
+    json_cache = dict() # key line number, value json data
```

Comment:
Style nit: I'm used to seeing `{}` instead of `dict()`.

I might also add a comment for `json_cache` mentioning that this is deliberately a shared cache among all instances of Parser subclasses. Otherwise having a class variable like this looks like it could be a mistake, since instance variables are much more common.

Reply:
- Style nit: in google python style guide, it recommends using data factory function instead of [], (), {}. Its more clear. `{}` could stand for set: `{1, 2, 3}` or dict: `{"a": 1, "b": 2}`, `()` could stand for tuple `(1, 2, 3)` or generator `(i for i in [1, 2, 3])`
- Agree, since its a temp solution and only used for lambda, it should be fine.